### ISSUE TITLE
Image info not being returned on require

### ISSUE THREAD

**Issue by xedef** [facebook/react-native#11261]
### Description

According to the documentation in [Why Not Automatically Size Everything?](https://facebook.github.io/react-native/docs/images.html#why-not-automatically-size-everything), requiring a local image should return an object containing information about the image but all I get is an integer for the image handle.

### Reproduction

```
import React, { Component } from 'react';
import {
  AppRegistry,
  View
} from 'react-native';

var i = require('./img.png')

export default class ReactTest extends Component {
  render() {

    console.log('Required image: ', JSON.stringify(i));

    return (
      <View />
    );
  }
}

AppRegistry.registerComponent('ReactTest', () => ReactTest);
```

```console.log``` call should return something like ```{"__packager_asset":true,"uri":"img.png","width":200,"height":300}``` but instead it just returns an integer.

### Additional Information

* React Native version: 0.39.0
* Platform: Android and iOS
* Operating System: macOS

**Comment by damusnet** [comment]
@facebook-github-bot expected

**Comment by damusnet** [comment]
If you take a look at the [source code](https://github.com/facebook/react-native/blob/master/Libraries/Image/Image.ios.js#L160) you'll see that the shape of a source is either the full object or an integer. In the case of a local require, you get a resource ID from the build tool, and the rest is abstracted away. Also, while I agree this is confusing, the docs say "the result **might** be". If you need more control over your image, you can use the `onLoad` [events](https://facebook.github.io/react-native/docs/image.html#onload), and the `getSize` [function](https://facebook.github.io/react-native/docs/image.html#getsize).

**Comment by Swordsman-Inaction** [comment]
@damusnet I have a question about images. I read the source code that `AssetRegistry` will return the image info, then the `ReactImageView` on Android will use an `ImageRequest` to get the real image, but I didn't go further. Just a simple question, is there a useful Util(both android and ios) I can use to get the real image(drawable or bitmap) in my own native module through the `require('XXX.png')` passed from JS code.

**Comment by damusnet** [comment]
@Swordsman-Inaction I'm sorry, I really don't know enough about RN internals to answer your question.

**Comment by Swordsman-Inaction** [comment]
@damusnet  I see. Thanks anyway.

**Comment by ragamufin** [comment]
I've tried to use `Image.getSize()` on an image that is retrieved using `require` however because `require` returns an integer I get the error "JSON value '8' of type NSNumber cannot be converted to a valid URLRequest". I've tried to use `findNodeHandle` on the integer but it simply returns the same integer. Other than doing `onLoad` in an image is there another way to get the size of the image dynamically in code?
